### DISCUSSION TITLE
v1.2.2: Fix optional sensor handling

### DISCUSSION THREAD

**Discussion by vmatare** [vmatare/thinkfan#135] [General]
This is a maintenance release that fixes a long-standing bug in handling sensors that may disappear at runtime (i.e. those marked with `optional: true`). An error in the logic surrounding that led to the `optional:` setting effectively being ignored.

Leave a comment to report whether this release fixes optional sensors for you.

<hr /><em>This discussion was created from the release <a href='https://github.com/vmatare/thinkfan/releases/tag/1.2.2'>Fix optional sensor handling</a>.</em>

**Comment by harmlessdrudge** [comment]
I was bitten by sensors changing ID (hwmon2 and hwmon3 interchanging) when setting up a Thinkpad X220 for a friend (running Linux Mint) -- happened only once. Set him up with v0.99. Would have used a later release but couldn't find a compiled version. (Not everyone has time or know-how to get set up for one small compilation.)
Thought that had happened again recently but friend now has some other problem with fan not starting automatically as it used to. I tried to fix it with 
```
[Unit]
Description=simple and lightweight fan control program
After=syslog.target

[Service]
Type=forking
EnvironmentFile=-/etc/default/thinkfan
ExecStartPre=/sbin/modprobe thinkpad_acpi
ExecStart=/usr/sbin/thinkfan $DAEMON_ARGS
PIDFile=/var/run/thinkfan.pid
ExecReload=/bin/kill -HUP $MAINPID

[Install]
WantedBy=multi-user.target
```
but this didn't work and I haven't had time to futz with it since. Some systemd info would be a help.

**Reply by vmatare** [reply]
What do you mean by "fan not starting automatically"? You mean thinkfan isn't starting on system boot? Or is it starting but the fan isn't behaving as expected? In any case, there should be an error message in your system log. Try `journalctl -el -u thinkfan` or run thinkfan manually on the command line with `thinkfan -n -vv`. Post the output and we might find out where your problem is.
